pre_commit_hooks/pretty_format_json.py

```diff
@@ -116,10 +117,10 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
                 if args.autofix:
                     _autofix(json_file, pretty_contents)
                 else:
-                    print(
-                        get_diff(contents, pretty_contents, json_file),
-                        end='',
+                    diff_output = get_diff(
+                        contents, pretty_contents, json_file,
                     )
+                    sys.stdout.buffer.write(diff_output.encode('utf-8'))
 
                 status = 1
         except ValueError:
```

Comment:
the other problem with how this is written is there's wayyy too much in this `try:` `except:` -- part of the fix should be to narrow the scope to only the raisable code (which was hiding this bug)